### PR TITLE
Mwawrzos/fix torch inductor pytorch to t

### PR DESCRIPTION
**Pull request by mwawrzos**
# What does this PR do ?

Support for the latest PyTorch version, which upgraded to triton 3.0.0.

**Collection**: requirements

# Changelog 
- Update requirements,
- Update guards accordingly,
- Update outdated API used by the guards.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Jenkins CI
To run Jenkins, a NeMo User with write access must comment `jenkins` on the PR.

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Comment by mwawrzos** [comment]
jenkins

**Comment by mwawrzos** [comment]
jenkins

File: nemo/collections/nlp/modules/common/megatron/attention.py [hunk 1]
```diff
@@ -67,11 +67,11 @@
 
 try:
     # Flash Attention Triton
-    import pkg_resources
+    from importlib.metadata import version
     from flash_attn.flash_attn_triton import flash_attn_func as flash_attn_func_triton
 
     # pinned triton version for flash-attention triton https://github.com/HazyResearch/flash-attention/blob/main/flash_attn/flash_attn_triton.py#L3
-    assert pkg_resources.get_distribution("triton").version == '2.0.0.dev20221202'
+    assert version("pytorch-triton") == '2.0.0.dev20221202'
```
Comment (ShriyaPalsamudram) [inline nemo/collections/nlp/modules/common/megatron/attention.py:74]: This assertion fails when latest pytorch-trition is installed based on requirements.txt

File: nemo/collections/nlp/modules/common/megatron/attention.py [hunk 2]
```diff
@@ -67,11 +67,11 @@
...
 
 except (ImportError, ModuleNotFoundError, AssertionError, pkg_resources.DistributionNotFound):
```
Comment (ShriyaPalsamudram) [inline nemo/collections/nlp/modules/common/megatron/attention.py:76]: Fails due to `NameError: name 'pkg_resources' is not defined`

**Comment by thomasdhc** [comment]
Thanks so the fix, launched e2e testing. I'll let you know the results

**Comment by mwawrzos** [comment]
jenkins

**Comment by mwawrzos** [comment]
closing in favor of https://github.com/NVIDIA/NeMo/pull/8805

**Pull request closed without merging**